drivers: nrf: uarte: check if RXSTARTED before issue STOPRX

rerickson1 opened the pull request:
Only issue STOPRX is RX has started otherwise we will get stuck in while loop.

rerickson1 commented:
@carlescufi  FYI.  Looks like something stalled with the checks. Is there a way to restart?

carlescufi commented:
@rerickson1 actually not sure. Try rebasing onto the latest master and force-pushing see if that fixes it.

anangl commented on drivers/serial/uart_nrfx_uarte.c:
```diff
@@ -1338,11 +1338,14 @@ static void uarte_nrfx_set_power_state(struct device *dev, u32_t new_state)
-		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPRX);
-		while (!nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_RXTO)) {
-			/* Busy wait for event to register */
+		if (nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_RXSTARTED)) {
```
It would be good to also clear the `RXSTARTED` event here, so that the next time this code is executed, it will check if RX has been actually started again.

rerickson1 committed:
drivers: nrf: uarte: check if RXSTARTED before issue STOPRX

Only issue STOPRX is RX has started.
This prevents getting stuck in while loop.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

rerickson1 commented:
@anangl Could use another review here.
